src/clj/game/cards-ice.clj

```diff
@@ -336,9 +337,10 @@
                                       (->> (vec (.split (:subtype card) " - "))
                                            (concat [target])
                                            (join " - "))))
+                      (update! (assoc card :subtype-target target))
```

Comment:
Found the problem. The `update!` changes the card as stored in `state` (signified by the `!`).
Now above this line the card is also `update!`ed to have the new `:subtype`. The `:subtype-target` `update!` overwrites this change (as you are `assoc`ing with the old `card`), which is why the tests fail (the card never actually morphs).

This works for me:
```clojure
(defn morph [state side card new old]
  (update! state side (assoc card
                        :subtype-target new
                        :subtype (->> (remove #(= old %) (.split (:subtype card) " - "))
                                      vec (concat [new]) distinct (join " - "))))
  (update-ice-strength state side card))
```